Fix error when purchasing subscriptions with Link

annemirasol opened the pull request:
Also fixes failing e2e tests, as noted in p1745251423830199-slack-C055WHLA98D

## Changes proposed in this Pull Request:
When purchasing subscriptions and the shopper uses Link, the purchase may because of payment token-related errors. 

Details:
- The payment will fail if the credit card stored in Link has never been used before. This happens because `WC_Stripe_UPE_Payment_Method_CC::create_payment_token_for_user()` is called, with no card details available, since the Stripe considers the payment method as type `link`.

## Testing instructions
1. Enable Link as a payment method.
2. In an incognito browser as the shopper, add a subscription to your cart.
3. Enter your credit card details. 
4. Make sure `Save your info for secure 1-click checkout with Link` is checked, and enter an email address and phone number, like the screenshot below.
5. In `develop` the payment fails, with below fatal error in logs. Subscriptions e2e tests also fail.
6. In this branch, the payment should succeed. Verify in **My Account** that you see the saved Link payment method.
8. Subscriptions e2e tests should also succeed.


<img width="846" alt="Screenshot 2025-04-21 at 1 17 22 PM" src="https://github.com/user-attachments/assets/38a901c1-2093-4a97-9e98-b701e722f6ae" />

Fatal in `develop`:
```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to WC_Stripe_Payment_Token_CC::set_fingerprint() must be of the type string, null given, called in /var/www/html/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-upe-payment-method-cc.php on line 98 and defined in /var/www/html/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-tokens/trait-wc-stripe-fingerprint.php:27
```


<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Please follow the following guidelines when writing testing instructions:

- Include screenshots if there is no similar flow in the critical flows: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Critical-flows
- Assume instructions will be copied over to the Release Testing Instructions wiki page: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions
- Assume instructions will be followed by external testers.
- Assume tester does not have intimate knowledge of Stripe.
-->

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Tested on mobile (or does not apply)

### Changelog entry

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->
Fixing an unreleased regression. Per recent discussions, we will no longer add such fixes to the changelog. 

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

Commit:
Fix error when saving Link payment method

Commit:
Add changelog and readme entries

Commit:
Use existing payment method instance instead of creating new

Mayisha commented on includes/payment-methods/class-wc-stripe-upe-payment-gateway.php:
```diff
@@ -1755,7 +1755,7 @@ public function retry_after_error( $response, $order, $retry, $force_save_source
 
 		sleep( $this->retry_interval );
-		$this->retry_interval++;
+		++$this->retry_interval;
```
QQ: same question as https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4244/files#r2053325952

annemirasol replied:
We [recently updated phpcs](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4207) and this was just the formatter doing its thing, as [WordPress coding standards prefer pre-increment/decrement operators](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#increment-decrement-operators).

Mayisha reviewed (approved):
Changes looks good to me 👍 

I got the following fatal error in`develop` branch when purchasing a subscription with Link-

<img width="1286" alt="Screenshot 2025-04-22 at 12 02 46 PM" src="https://github.com/user-attachments/assets/9e5031b1-3110-4ee4-894f-b1a063321f32" />

I have confirmed that in this branch subscription purchase is working well by testing the following sceanrios-

✅ Purchased a subscription with Link successfully.
✅ The Link payment method got saved in the shopper's account.
✅ Renewed the subscription as an admin.
✅ Purchased a simple product with the saved Link method.
✅ Purchases a simple product with a regular card with `save payment method` checkbox selected. The card was saved in the shopper's account.

Commit:
Merge branch 'develop' into fix/link-payment-tokens

Commit:
Restore SPE support

Commit:
Remove unnecessary changelog and readme entries

Commit:
Merge branch 'develop' into fix/link-payment-tokens

annemirasol commented:
Additional changes:
- I realized I broke saved payment methods for SPE (non-card), so I added support for it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4246/commits/cfedf82d760784e3838618dab1746b40b11caabc
- Per recent discussions, I removed the readme and changelog entries.

I have tested the following after making these changes:
✅ Saving payment methods in My Account (CC, Link, CashApp)
✅ Saving payment methods during checkout (CC, Link, CashApp)
✅ SPE: Saving payment methods in My Account (CC, CashApp)
✅ SPE: Saving payment methods in checkout (CC, CashApp)

Commit:
e2e: Use check() instead of click() for checkbox

Commit:
Add SPE e2e tests to default tests' testIgnore

Commit:
e2e: Try using force: true with .check()
